lib/api/BackbeatAPI.js

```diff
@@ -281,12 +310,47 @@ class BackbeatAPI {
     /**
      * Query StatsClient for all ops given
      * @param {array} ops - array of redis key names to query
+     * @param {string} site - site name or '*' wildcard
      * @param {function} cb - callback(err, res)
      * @return {undefined}
      */
-    _queryStats(ops, cb) {
+    _queryStats(ops, site, cb) {
         return async.map(ops, (op, done) => {
-            this._statsClient.getStats(this._logger, op, done);
+            const queryString = `${op}:${site}`;
+            // Query all sites if given wildcard
+            if (queryString.indexOf('*') >= 0) {
+                return this._redisClient.scan(queryString, undefined,
+                (err, res) => {
+                    if (err) {
+                        // escalate error to log later
+                        return done({
+                            message: `Redis error: ${err.message}`,
+                            type: errors.InternalError,
+                            method: 'BackbeatAPI._queryStats',
+                        });
+                    }
+                    const allKeys = res.map(key => {
+                        const arr = key.split(':');
+                        // Remove the "requests:<timestamp>" and process
+                        return arr.slice(0, arr.length - 2).join(':');
+                    });
+                    const reducedKeys = [...new Set(allKeys)];
```

Comment:
I like this construct to uniquify an array :+1: 

Though is the `scan` function prone to returning duplicate keys in `res`?

Reply:
Yes there is a possibility `scan` will return duplicate keys